go/libkb/globals.go

```diff
@@ -508,17 +508,27 @@ func (g *GlobalContext) FlushCaches() {
 	g.configureMemCachesLocked(true)
 }
 
-func (g *GlobalContext) configureDiskCachesLocked() error {
+func (g *GlobalContext) configureDiskCachesLocked() (err error) {
 	// We consider the local DBs as caches; they're caching our
 	// fetches from the server after all (and also our cryptographic
 	// checking).
 	g.LocalDb = NewJSONLocalDb(NewLevelDb(g, g.Env.GetDbFilename))
 	g.LocalChatDb = NewJSONLocalDb(NewLevelDb(g, g.Env.GetChatDbFilename))
 
 	epick := FirstErrorPicker{}
-	epick.Push(g.LocalDb.Open())
-	epick.Push(g.LocalChatDb.Open())
-	return epick.Error()
+	epick.Push(g.LocalDb.ForceOpen())
+	epick.Push(g.LocalChatDb.ForceOpen())
+	if err = epick.Error(); err != nil {
+		// Failed to open leveldb. If we are configured such that using the in-memory cache is an
+		// option, then let's sub that in now.
```

Comment:
is it possible one worked and the other didn't?  i.e. no error with LocalDb but and error with LocalChatDb?

Reply:
I'm not sure how, I thought it would be simpler though if they were all on the same backend.